Collector/PluginClientFactorySubscriber.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Http\HttplugBundle\Collector;
+
+use Http\Client\Common\PluginClientFactory;
+use Http\HttplugBundle\Collector\PluginClientFactory as CollectorPluginClientFactory;
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * This subscriber ensures that every PluginClient created when using Http\Client\Common\PluginClientFactory without
+ * using the Symfony dependency injection container uses the Http\HttplugBundle\Collector\PluginClientFactory factory
+ * when profiling is enabled. This allows 0 config profiling of third party libraries which use HTTPlug.
+ *
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ *
+ * @internal
+ */
+final class PluginClientFactorySubscriber implements EventSubscriberInterface
```

Comment:
i heard such a class should still be called ...Listener. it semantically is an event listener. the subscriber interface is about how to wire up the listener and less central.

Reply:
Right, I updated the name.